src/libraries/System.Private.CoreLib/src/System/ArraySegment.cs

```diff
@@ -276,6 +279,41 @@ private void ThrowInvalidOperationIfDefault()
             }
         }
 
+        /// <summary>
+        /// For <see cref="ArraySegment{Char}"/>, returns a new instance of string that represents the characters pointed to by the segment.
+        /// Otherwise, returns a <see cref="string"/> with the name of the type and the number of elements.
+        /// </summary>
+        public override string ToString()
```

Comment:
Using `ToString` is an observable behavior change. The common approach is using a private property called `DebuggerDisplay`.